website/content/roadmap.md

```diff
@@ -15,77 +15,133 @@ url: roadmap
+schema of a database.
 
-<--->
+#### Query transparency
 
-## Alternative backends
+PRQL's compiler already contains structured data about the query. We'd like to
+offer transparency to tools which use PRQL, so they can offer lineage
+information, such as which tables are queried, and a DAG of transformations for
+each column.
 
-Currently, PRQL only transpiles into SQL. It could be much more powerful (and in
-some cases performant) if we develop a data-frame-handling-library backend. To
-be more precise, we would want to apply PRQL's AST to a in-memory dataframe of a
-performance-optimized library (such as [Polars](https://www.pola.rs/)).
+{{< /columns >}}
+
+## Long term
+
+{{< columns >}}
+
+#### SQL-to-PRQL conversion
+
+While PRQL already allows for a gradual on-ramp — there's no need to switch
+everything to PRQL right away — it would also be useful to be able to convert
+existing SQL queries to PRQL, rather than having to rewrite them manually. For
+many queries, this should be fairly easy. (For some it will be very difficult,
+but we can start with the easy ones...)
+
+#### Language
+
+While the core semantics and syntax of the language are now fairly stable, we
+are planning
+[a few major features](https://github.com/PRQL/prql/issues?q=is%3Aopen+is%3Aissue+label%3Amajor-feature+label%3Alanguage-design)
+that will give PRQL the feeling of a real programming language and elevate it in
+[the chomsky hierarchy](https://en.wikipedia.org/wiki/Chomsky_hierarchy).
+Honorable mentions here are recursive CTEs (or rather functions), algebraic type
+system, pre-specified join conditions and regex.
+
+Note that these features will probably inflict breaking changes with each minor
+release before we stabilize the 1.0, the first indefinitely supported language
+edition.
 
-This would allow data scientists, analysts and general Python developers to
-transform DataFrames with PRQL queries. One language for all data
-transformations!
+<--->
+
+#### Alternative backends
+
+Currently, PRQL only transpiles into SQL, using connectors such as DuckDB to
+access other formats, such as Pandas dataframes. But PRQL can be much more
+general than SQL — we could directly compile to any relational backend, offering
+more flexibility and performance — and a consistent experience for those who use
+multiple tools.
 
-## PRQL as a tool
+For example, we could compile PRQL to RQ (Relational Query intermediate
+representation) and then use that to apply the transformations to an in-memory
+dataframe of a performance-optimized library (such as
+[Polars](https://www.pola.rs/)) or a Google Sheets spreadsheet. Alternatively,
+we could even convert RQ to [Substrait](https://substrait.io/).
 
-`pyprql` is a step into direction of a general data handling program. Building
-on this, we want to build a tool that can read many data sources, offers syntax
-highlighting, auto-complete and type-inference using information from database's
-schema.
+#### PRQL as a tool
```

Comment:
@snth do you want to mention prql-query? I would but I know you've been suggesting we delay